src/lib/compression/zstd_header_check.h

```diff
@@ -0,0 +1,35 @@
+/* This code is subject to the terms of the Mozilla Public License, v.2.0. http://mozilla.org/MPL/2.0/. */
+#pragma once
+
+#include "zstd/zstd.h"
+#include <string>
+
+namespace cimbar {
+
+class zstd_header_check
+{
+public:
+	static std::string get_filename(const unsigned char* data, size_t len)
+	{
+		if (!ZSTD_isSkippableFrame(data, len))
+			return "";
+
+		std::string res;
+		res.resize(500, '\0');
+		size_t sz = ZSTD_readSkippableFrame(res.data(), res.size(), nullptr, data, len);
+		if (sz <= 1)
+			return "";
+
+		switch (res[0])
+		{
+			case 1:
```

Comment:
It's somewhat moot at the moment, but for future proofing we're treating the first byte as a version number. For now, it's:
`01 <filename bytes>`

We have some flexibility -- zstd has a "magic value" attached to all skippable frames, that could be used for the same purpose. But using that value locks us into having to decode it, vs a potential string match to identify the start of the frame.